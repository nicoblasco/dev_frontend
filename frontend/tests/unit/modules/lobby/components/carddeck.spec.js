import {shallowMount} from '@vue/test-utils'
import CardDesk from '@/modules/lobby/components/CardDeck'

describe('Unit test en CardDesk', () => {
    
    test('debe de renderizar el componente correctamente ', () => {

        const wrapper =  shallowMount(CardDesk)

        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostrar tantos botones como cartas validas contenga ', () => {
        const wrapper =  shallowMount(CardDesk)
        const validVotes= [1,2,3,5,8,13,20,40,'?']
        const buttons = wrapper.findAll('button').length
        expect (validVotes.length).toEqual(buttons);
    })
})