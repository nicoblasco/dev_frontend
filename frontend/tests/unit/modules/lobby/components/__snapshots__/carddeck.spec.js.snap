// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test en CardDesk debe de renderizar el componente correctamente  1`] = `
<div>
  <div class="row text-center h-100">
    <p class="deck-text">Choose your card ðŸ‘‡</p>
  </div>
  <div class="row text-center">
    <div class="col-md-12 align-self-center"><button class="btn btn-outline-primary card-player" disabled="">1</button><button class="btn btn-outline-primary card-player" disabled="">2</button><button class="btn btn-outline-primary card-player" disabled="">3</button><button class="btn btn-outline-primary card-player" disabled="">5</button><button class="btn btn-outline-primary card-player" disabled="">8</button><button class="btn btn-outline-primary card-player" disabled="">13</button><button class="btn btn-outline-primary card-player" disabled="">20</button><button class="btn btn-outline-primary card-player" disabled="">40</button><button class="btn btn-outline-primary card-player" disabled="">?</button></div>
  </div>
</div>
`;
